### PR TITLE
Ncnews 6

### PR DESCRIPTION
**Pull request by baloglu**
This PR:
-Adds a new endpoint /api/articles/:article_id/comments that return comments for the given article
- Adds the relevant tests

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -91,4 +91,30 @@ describe("API endpoints", () => {
...
+            .then((result) => {
+                const comments = result.body.comments;
+                const expected = 11
+                expect(comments.length).toBe(expected)
+                expect(comments[0]).toHaveProperty('comment_id')
+                expect(comments[0]).toHaveProperty('votes')
+                expect(comments[0]).toHaveProperty('created_at')
+                expect(comments[0]).toHaveProperty('author')
+                expect(comments[0]).toHaveProperty('body')
+                expect(comments[0]).toHaveProperty('article_id')
```
Comment (MKD01) [inline __tests__/api.test.js]: This is only checking if the first element in the comments array contains these properties, but it should be checking that every element contains these properties. 

We also know what the value of article_id should be for each of these comments, so this should be included in the above assertion as well.

File: __tests__/api.test.js [hunk 2]
```diff
@@ -91,4 +91,50 @@ describe("API endpoints", () => {
...
+    test("When article id for comments is not a number it should return syntax error", () => {
+        return request(app)
+        .get("/api/articles/hundred/comments")
+        .expect(400)
+            .then((result) => {
+                const message = JSON.parse(result.error.text).msg;
+                const expected = 'Invalid input syntax'
+                expect(message).toEqual(expected)
+        });
+    })
```
Comment (MKD01) [inline __tests__/api.test.js:155]: Another test to consider is what if a valid article has no comments? What should this test look like?

File: __tests__/api.test.js [hunk 3]
```diff
@@ -91,4 +91,50 @@ describe("API endpoints", () => {
...
+    test("When article id doesn't exist for comments it should return an error", () => {
+        return request(app)
+        .get("/api/articles/100/comments")
+        .expect(400)
+            .then((result) => {
+                const message = JSON.parse(result.error.text).msg;
+                const expected = 'There are no comments for this article'
+                expect(message).toEqual(expected)
+        });
+    })
```
Comment (MKD01) [inline __tests__/api.test.js]: Should this be a 400 status error?

**Review by MKD01 — CHANGES_REQUESTED** [review]
Good work so far, but there are a few things to change/add, once you have made all your changes push them up to this branch and then re-submit a review request via /nchelp pr

**Review by Beth-McCormack — APPROVED** [review]

**Pull request merged**